src/NuGet.Core/NuGet.Configuration/PackageNamespaces/PackageNamespacesConfiguration.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NuGet.Configuration
+{
+    public class PackageNamespacesConfiguration
+    {
+        /// <summary>
+        /// Source name to package namespace list.
+        /// </summary>
+        public Dictionary<string, IReadOnlyList<string>> Namespaces { get; }
+
+        public PackageNamespacesConfiguration(Dictionary<string, IReadOnlyList<string>> namespaces)
```

Comment:
It seems you only want to create instances of this class using `GetPackageNamespacesConfiguration(ISettings)`. If that's the case, can we make this constructor less visible?

```suggestion
        protected PackageNamespacesConfiguration(Dictionary<string, IReadOnlyList<string>> namespaces)
```

Reply:
Great point. Fixed.